scrapy/core/downloader/tls.py

```diff
@@ -0,0 +1,16 @@
+from OpenSSL import SSL
+
+
+METHOD_SSLv3 = 'SSLv3'
+METHOD_TLS = 'TLS'
+METHOD_TLSv10 = 'TLSv1.0'
+METHOD_TLSv11 = 'TLSv1.1'
+METHOD_TLSv12 = 'TLSv1.2'
+
+openssl_methods = {
+    METHOD_TLS:    SSL.SSLv23_METHOD,                   # protocol negotiation (recommended)
```

Comment:
This is similar to OpenSSL (yet to come) 1.1:
https://www.openssl.org/docs/manmaster/ssl/SSL_CTX_new.html

```
 #define SSLv23_method           TLS_method
 #define SSLv23_server_method    TLS_server_method
 #define SSLv23_client_method    TLS_client_method
```

> TLS_method(), TLS_server_method(), TLS_client_method()
> These are the general-purpose version-flexible SSL/TLS methods. The actual protocol version used will be negotiated to the highest version mutually supported by the client and the server. The supported protocols are SSLv3, TLSv1, TLSv1.1 and TLSv1.2. Most applications should use these method, and avoid the version specific methods described below.

> SSLv23_method(), SSLv23_server_method(), SSLv23_client_method()
> Use of these functions is deprecated. They have been replaced with the above TLS_method(), TLS_server_method() and TLS_client_method() respectively. New code should use those functions instead.

Also see https://www.openssl.org/news/changelog.txt

```
Changes between 1.0.2f and 1.1.0  [xx XXX xxxx]
(...)
  *) Version negotiation has been rewritten. In particular SSLv23_method(),
     SSLv23_client_method() and SSLv23_server_method() have been deprecated,
     and turned into macros which simply call the new preferred function names
     TLS_method(), TLS_client_method() and TLS_server_method(). All new code
     should use the new names instead. Also as part of this change the ssl23.h
     header file has been removed.
     [Matt Caswell]
```

I would not want us to use `SSLv23` as a setting default so as no to confuse users that it's SSL only.
But I'm open for other names for the default protocol flexible setting.